### PR TITLE
Bump actions to Node.js 20 versions

### PR DESCRIPTION
**Pull request by patrick-rivos**
Fixes list of warning on runs. Changes done via find & replace-all 
https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

### CODE REVIEW

File: .github/workflows/generate-summary.yaml [hunk 1]
```diff
@@ -337,7 +337,7 @@ jobs:
               body: 'Bisecting with #${{ needs.generate-issues.outputs.new_issue_num }}'
```
Comment (ewlu) [inline .github/workflows/generate-summary.yaml:337]: need to update actions/github-script@v6 as well https://github.com/patrick-rivos/gcc-postcommit-ci/pull/1006/files#diff-212c3469e283d0c467d86f01ac4ab899a704631cdf26e87daf5d75bb9abcf679R330

File: .github/workflows/generate-summary.yaml [hunk 2]
```diff
@@ -454,23 +454,23 @@ jobs:
...
           path: |
             riscv-gnu-toolchain/current_logs.zip
           retention-days: 90
 
       - name: Upload previous log failures
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: ${{ inputs.prefix }}${{ steps.bisection-hash.outputs.orig_gcchash }}-regenerated-previous-logs
           path: |
```
Comment (ewlu) [inline .github/workflows/generate-summary.yaml:476]: may also want to update the JasonEtco/create-an-issue@v2 in regenerate summary https://github.com/patrick-rivos/gcc-postcommit-ci/pull/1006/files#diff-212c3469e283d0c467d86f01ac4ab899a704631cdf26e87daf5d75bb9abcf679R509

File: .github/workflows/deploy-dashboard.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ jobs:
       id-token: write
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup env
         uses: ./.github/actions/common/setup-env
```
Comment (ewlu) [inline .github/workflows/deploy-dashboard.yaml:31]: have some other scripts to double check here https://github.com/patrick-rivos/gcc-postcommit-ci/blob/main/.github/workflows/deploy-dashboard.yaml#L73-L90

File: .github/workflows/init-submodules.yaml [hunk 4]
```diff
@@ -130,7 +130,7 @@ jobs:
       # Artifacts are reliable but ~30 min slower to set up.
```
Comment (ewlu) [inline .github/workflows/init-submodules.yaml:130]: also update actions/cache/save@v3 https://github.com/patrick-rivos/gcc-postcommit-ci/pull/1006/files#diff-bc25712c7e944a3dc493a2b5f6fb10d56d41e807d4357c010af07cf641b24720L112

**Review by ewlu — CHANGES_REQUESTED** [review]
missed a couple of actions

File: .github/workflows/generate-summary.yaml (hunk 2 above)
Comment (patrick-rivos) [reply .github/workflows/generate-summary.yaml:476]: JasonEtco/create-an-issue@v3 does not seem to exist:
https://github.com/JasonEtco/create-an-issue

File: .github/workflows/generate-summary.yaml (hunk 1 above)
Comment (patrick-rivos) [reply .github/workflows/generate-summary.yaml:337]: Done

File: .github/workflows/deploy-dashboard.yaml (hunk 3 above)
Comment (patrick-rivos) [reply .github/workflows/deploy-dashboard.yaml:31]: EndBug/add-and-commit@v9 does not have a v10
actions/deploy-pages@v2 Updated to v4
actions/configure-pages@v3 seems fine?
actions/upload-pages-artifact@v2 seems fine?

Comment (ewlu) [reply .github/workflows/deploy-dashboard.yaml:31]: actions/configure-pages looks like theres a v5 https://github.com/actions/configure-pages/releases. the README hasn't been updated in 2 years but the action was recently updated to use node 20 (https://github.com/actions/configure-pages/commit/3a014131e99a7bf2daf896787a8f92789126d038)

actions/upload-pages-artifact looks like theres a v4 https://github.com/actions/upload-pages-artifact/releases/tag/v3.0.0. Not sure what the difference is bc the changelog doesn't explicitly say it changes to use node 20

**Review by ewlu — APPROVED** [review]

**Pull request merged**